python/lancedb/fts.py

```diff
@@ -98,6 +113,32 @@ def populate_index(index: tantivy.Index, table: LanceTable, fields: List[str]) -
     return row_id
 
 
+def resolve_path(schema, field_name: str) -> pa.Field:
+    """
+    Resolve a nested field path to a list of field names
+
+    Parameters
+    ----------
+    field_name : str
+        The field name to resolve
+
+    Returns
+    -------
+    List[str]
+        The resolved path
+    """
+    path = field_name.split(".")
+    if len(path) == 1:
+        return False
+    tbl = schema.empty_table()
+    for _ in range(len(path) - 1):
+        prev = tbl
+        tbl = prev.flatten()
+        if prev == tbl:
+            break
+    return tbl.schema.field(field_name)
```

Comment:
Can we make this handle one-level and also not create temporary tables?

```suggestion
    path = field_name.split(".")
    field = schema.field(path.pop(0))
    for segment in path:
        if isinstance(field.data_type, pa.StructType):
            field = field.data_type.field(segment)
        else:
            raise KeyError(f"field {field_name} not found in schema {schema}")
    return field
```

Reply:
thanks. needs to be `field.type` and I changed it to use `pa.types.is_struct`.